.travis.yml

```diff
@@ -11,6 +11,12 @@ after_success: "[[ $TRAVIS_PULL_REQUEST == \"false\" ]] && mvn deploy"
 
 sudo: false
 
+# https://github.com/travis-ci/travis-ci/issues/3259
+addons:
+  apt:
+    packages:
+      - oracle-java8-installer
```

Comment:
Indeed this is necessary for the build to succeed. With JDK 8u31, the
sources won't compile, since the compiler fails to infer some types. Either
I'll have to rework the PR and introduce some boilerplate, or we use a
newer JDK.

The issue in the comment states other, possibly faster ways to get a newer
JDK, for instance to use a newer Ubuntu version. Shall I investigate
further?